File: original_recipe/bbcursive/src/main/java/bbcursive/lib/chlit_.java
package bbcursive.lib;

import bbcursive.func.UnaryOperator;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.nio.ByteBuffer;

/**
char literal
 */
public class chlit_ {
    @NotNull
    public static UnaryOperator<ByteBuffer> chlit(char c) {
        return new ByteBufferUnaryOperator(c);
    }

    @NotNull
    public static UnaryOperator<ByteBuffer> chlit(@NotNull CharSequence s) {
        return chlit(s.charAt(0));
    }


    private static class ByteBufferUnaryOperator implements UnaryOperator<ByteBuffer> {
        private final char c;

        ByteBufferUnaryOperator(char c) {
            this.c = c;
        }

        @NotNull
        @Override
        public String toString() {
            return "c8'" +
                    c+"'";
        }

        @Nullable
        @Override
        public ByteBuffer invoke(@Nullable ByteBuffer buf) {
            if (null == buf) {
                return null;
            }
            if (buf.hasRemaining()) {
                byte b = buf.get();
                return (c & 0xff) == (b & 0xff) ? buf : null;
            }
            return null;



        }
    }
}

File: original_recipe/bbcursive/src/main/java/bbcursive/lib/chlit_.kt
package bbcursive.lib

import bbcursive.func.UnaryOperator
import java.nio.ByteBuffer

/**
 * char literal
 */
object chlit_ {
      @JvmStatic
  fun chlit(c: Char): UnaryOperator<ByteBuffer?> {
        return ByteBufferUnaryOperator1(c)
    }

    @JvmStatic
    fun chlit(s: String): UnaryOperator<ByteBuffer?> {
        return chlit(s[0])
    }

    private class ByteBufferUnaryOperator1(private val c: Char) : UnaryOperator<ByteBuffer?> {
        override fun toString(): String {
            return "c8'" +
                    c + "'"
        }

        override fun invoke(buf: ByteBuffer?): ByteBuffer? {
            if (null == buf) {
                return null
            }
            if (buf.hasRemaining()) {
                val b = buf.get()
                return if (c.code .toUByte() == b .toUByte()) buf else null
            }
            return null
        }
    }
}